askama/src/filters/builtin.rs

```diff
@@ -426,31 +427,265 @@ const _: () = {
     }
 };
 
-pub enum Pluralize<S, P> {
-    Singular(S),
-    Plural(P),
+/// Render `value` if it is not its "default" value, see [`DefaultFilterable`],
+/// otherwise `fallback`.
+#[inline]
+pub fn assigned_or<L: DefaultFilterable, R>(
+    value: &L,
+    fallback: R,
+) -> Result<Either<L::Filtered<'_>, R>, L::Error> {
+    match value.as_filtered()? {
+        Some(value) => Ok(Either::Left(value)),
+        None => Ok(Either::Right(fallback)),
+    }
+}
+
+/// A type (or a reference to it) that can be used in [`|assigned_or`](assigned_or).
+///
+/// The type is either a monad such as [`Option`] or [`Result`], or a type that has a well defined,
+/// trivial default value, e.g. an [empty](str::is_empty) [`str`] or `0` for integer types.
+#[diagnostic::on_unimplemented(
+    label = "`{Self}` is not `|assigned_or` filterable",
+    message = "`{Self}` is not `|assigned_or` filterable"
+)]
+pub trait DefaultFilterable {
```

Comment:
It could be worth adding a unit tests for the various implementations of this that captures each implementation. E.g.:

```rust
assert_eq!(0_u8.as_filtered(), Ok(None));
assert_eq!(1_u8.as_filtered(), Ok(Some(1));
```